src/posts/react-native-monorepo.md

```diff
@@ -0,0 +1,183 @@
+title: "Migrating to a Mobile Monorepo for React Native"
+published_on: May 29, 2017
+author: Jared Forsyth
+team: Mobile
+...
+
+Over the past few months, we've been adding some [React
+Native](https://github.com/facebook/react-native) to our existing
+[iOS](https://itunes.apple.com/us/app/khan-academy-you-can-learn-anything/id469863705?mt=8)
+and
+[Android](https://play.google.com/store/apps/details?id=org.khanacademy.android&hl=en)
+apps. We started out by just creating a `react-native` repository and having
+it be a submodule of our respective `ios` and `android` git repositories, but
+we quickly found that there was a fair amount of friction in keeping things
+coordinated between the three. We've now moved all of our mobile-related
+repositories (including a `mobile-scripts` and a `shared-webview` repository)
+into a single `mobile` [monorepo](https://danluu.com/monorepo/).
+
+## Why did we do it?
+
+When making changes to the js↔native FFI, we need to make the change to
+both the Android and the iOS codebases, or else we'll get runtime errors. With
+three repositories to work with, it was too easy to forget to add one or the
+other (as you're generally developing with just one simulator open), and then
+it breaks the build for other environment.
+
+We'd also get into a state where the `master` branch of `react-native`
+contained some changes that had only been coordinated with e.g. the `ios`
+repo, and the `android` repo's `react-native` submodule would be several
+commits behind. Then someone working on the android side would update the
+submodule, and they'd have to track down all the breakages.
+
+In short, we started running into a lot of synchronization issues, and we
+really wanted to just have things be in the same place, such that pull
+requests could be combined, reviews would be more coherent, and it would be
+easier to verify correctness between codebases.
+
+### Anticipated Pros
+
+- changing js↔native apis would be easier because you only need a single
+  pull request instead of 3
+- as a result, they would be less likely to get out of sync
+- not having to mess with submodules 🎉
+
+### Anticipated Cons
+
+- we might lose git history (this didn't turn out to be the case)
+- we'd have to change all of our jenkins build scripts
+- moving all our developers to a new repository requires coordination
+- we'd lose any in-flight branches & open PRs to the old repositories (we
+  actually found a solution for this too)
+
+## What were the steps?
+
+### Setup
+
+Make a fresh monorepo
+
+```sh
+mkdir mobile; cd mobile; git init .
+```
+
+Have all the repos that you want to combine cloned & fully up to date
```

Comment:
I put it in passive voice b/c I assume that these repos are already cloned -- because you're actively developing them.